Clarification for matching (#1808)
modelica/fmi-standard: docs/2_4_common_schema.adoc

_The structured naming convention of the <<ModelVariables>> is independent from the terminal names and member variable names._

_A tool may choose to connect terminals with a different or unknown `terminalKind`, if the `matchingRule` matches.

> 

Reviewer: ```suggestion
A tool may choose to take the structured naming convention of variable names into account when matching terminals.
This can for example be used to allow matching between terminals employing arrays mapped to scalar variables using structured naming convention, and those using native FMI 3.0 arrays.
```

> If a `TerminalMemberVariable` or `TerminalStreamMemberVariable` is an array, "matching" for the matching rules `plug` and `bus` includes "matching based on structured naming convention" besides pure name-equality if the following conditions are fulfilled: 
> 
> * The FMU uses the structured naming convention, i.e., *<<variableNamingConvention,`variableNamingConvention = structured`>>*.
> * The "naming convention array" member variables end with the array indices in squared brackets.
> * The syntax is `B-name [ arrayIndices ]`, where `B-name` is the "naming convention array" name (see the precise syntax in the structured naming convention).
> * The "naming convention array" may be 0 or 1 based.
> * All elements of the "naming convention array" must be present as scalar variables in consecutive order.
> * A ScalarVariable with the "naming convention array" name must not exist at the same time.
>   _[I.e. if ScalarVariable `x[1]` exists then ScalarVariable `x` must not exist).]_

Reviewer: ```suggestion
```

> Example: Assume that an (FMI 3.0) FMU has a one-dimensional array variable `x` of length 3 as member variable of a terminal and is connected to another (FMI 2.0 or FMI 3.0) FMU supporting structured naming convention, which has scalar variables `x[1]`, `x[2]`, `x[3]` as member variables of a terminal.
> Then `x[1]`, `x[2]`, `x[3]` are interpreted as a "naming convention array", which "matches" the array variable `x` of the first FMU]

Reviewer: ```suggestion
Assume that an (FMI 3.0) FMU has a one-dimensional array variable `x` of length 3 as member variable of a terminal and is connected to another (FMI 2.0 or FMI 3.0) FMU supporting structured naming convention, which has scalar variables `x[1]`, `x[2]`, `x[3]` as member variables of a terminal.
Then `x[1]`, `x[2]`, `x[3]` are interpreted as a "naming convention array" with name `x`, which "matches" the array variable `x` of the first FMU, in name-based matching.
Similarly the notional "naming convention array" `x` would participate in sequence based matching as a single notional variable in the place of the underlying scalar variables.
```

> 
>  An array ModelVariable in a terminal with matchingRule `sequence` may be interpreted as adding the scalar elements in the serialized form described in section <<serialization-of_variables>>.

Reviewer: ```suggestion
```
